libraries/chain/controller.cpp

```diff
@@ -2104,13 +2114,7 @@ struct controller_impl {
    }
 
    checksum256_type calculate_action_merkle() {
-      vector<digest_type> action_digests;
-      const auto& actions = pending->_block_stage.get<building_block>()._actions;
-      action_digests.reserve( actions.size() );
-      for( const auto& a : actions )
-         action_digests.emplace_back( a.digest() );
-
-      return merkle( move(action_digests) );
+      return merkle( move(pending->_block_stage.get<building_block>()._action_receipt_digests) );
```

Comment:
Let's just remove `calculate_action_merkle` and call `merkle( std::move( bb._action_receipt_digests ) )` from `finalize_block`.

Reply:
Nice, I like that, we are doing other moves there.

Reply:
Removed the `calculate_*_merkle` methods and call `merkle` directly in `finalize_block`.